paddlenlp/trainer/trainer.py

```diff
@@ -2780,6 +2780,12 @@ def evaluation_loop(
 
         # Metrics!
         if self.compute_metrics is not None and all_preds is not None and all_labels is not None:
+            if self.args.tensor_parallel_degree > 1 and all_preds.shape != all_labels.shape:
```

Comment:
```suggestion
            if self.args.tensor_parallel_degree > 1 and isinstance(all_preds, paddle.Tensor) all_preds.shape != all_labels.shape:
```

然后这里加一个注释吧，all_gather logits for tp